dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -83,14 +84,29 @@ public GHRepository getRepo(String repoName) throws IOException {
         return gitHubUtil.getRepo(repoName);
     }
 
-    public PagedSearchIterable<GHContent> findFilesWithImage(String image, String org) throws IOException {
+    public Optional<List<PagedSearchIterable<GHContent>>> findFilesWithImage(
+            String image,
+            Map<String, Boolean> orgsToIncludeOrExclude,
+            Integer gitApiSearchLimit) throws IOException {
         GHContentSearchBuilder search = gitHubUtil.startSearch();
         // Filename search appears to yield better / more results than language:Dockerfile
         // Root cause: linguist doesn't currently deal with prefixes of files:
         // https://github.com/github/linguist/issues/4566
         search.filename("Dockerfile");
-        if (org != null) {
-            search.user(org);
+        if (!orgsToIncludeOrExclude.isEmpty()) {
+            StringBuilder includeOrExcludeOrgsQuery = new StringBuilder();
+            for (Map.Entry<String, Boolean> org : orgsToIncludeOrExclude.entrySet()){
+                if (org.getKey() != null) {
+                    if (org.getValue()) {
+                        log.info("Including the org {} in the search.", org.getKey());
+                        includeOrExcludeOrgsQuery.append(String.format("user:%s ", org.getKey()));
```

Comment:
why not simply do `search.user(org.getKey())`?

Reply:
There can be multiple orgs that will need to be appended to the search query. From my understanding `search.user(org)` internally reslolves to `search.q(user:org)`. If we call call `search.user(org)` wouldn't the search query keep getting overwritten with every call and have only the last org added? Instead, using the `search.q` method allows us to construct the query string with all orgs.

Reply:
If you dig into `search.q()` you'll notice that it's just appending search terms to a `GHSearchBuilder`. If it were the case that it was overriding, you'd have a similar problem with `search.q()` here overriding `search.filename()` above.